src/math/interp.rs

```diff
@@ -26,15 +26,23 @@ impl TryFrom<&str> for InterpolateMode {
 
 #[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
 pub struct InterpRange<T> {
-    pub beg: T,
-    pub end: T,
+    beg: T,
+    end: T,
 }
 
 impl<T> InterpRange<T> {
     #[inline]
     pub fn new(beg: T, end: T) -> Self {
         Self { beg, end }
     }
+
+    pub fn begin(self) -> T {
+        self.beg
+    }
+
+    pub fn end(self) -> T {
+        self.end
+    }
 }
```

Comment:
There is no benefit to this

Reply:
Was there when I added an assert in new for `begin <= end`, I'll revert